pkg/resource/table/hooks.go

```diff
@@ -95,3 +102,242 @@ func isTableUpdating(r *resource) bool {
+	for i := range svcTags {
+		tags[i] = &v1alpha1.Tag{
+			Key:   svcTags[i].Key,
+			Value: svcTags[i].Value,
+		}
+	}
+	return tags
+}
+
+// svcTagsFromResourceTags transforms a *v1alpha1.Tag array to a *svcsdk.Tag array.
+func sdkTagsFromResourceTags(rTags []*v1alpha1.Tag) []*svcsdk.Tag {
+	tags := make([]*svcsdk.Tag, len(rTags))
+	for i := range rTags {
+		tags[i] = &svcsdk.Tag{
+			Key:   rTags[i].Key,
+			Value: rTags[i].Value,
+		}
+	}
+	return tags
+}
+
+// computeTagsDelta compares two Tag arrays and return three different list
+// containing the added, updated and removed tags.
+// The removed tags only contains the Key of tags
+func computeTagsDelta(
+	a []*v1alpha1.Tag,
+	b []*v1alpha1.Tag,
+) (added, updated []*v1alpha1.Tag, removed []*string) {
+	var visitedIndexes []string
+mainLoop:
+	for _, aElement := range a {
+		visitedIndexes = append(visitedIndexes, *aElement.Key)
+		for _, bElement := range b {
+			if equalStrings(aElement.Key, bElement.Key) {
+				if !equalStrings(aElement.Value, bElement.Value) {
+					updated = append(updated, bElement)
+				}
+				continue mainLoop
+			}
+		}
+		removed = append(removed, aElement.Key)
+	}
+	for _, bElement := range b {
+		if !ackutil.InStrings(*bElement.Key, visitedIndexes) {
+			added = append(added, bElement)
+		}
+	}
+	return added, updated, removed
+}
+
+func equalStrings(a, b *string) bool {
+	if a == nil {
+		return b == nil || *b == ""
+	}
+	return (*a == "" && b == nil) || *a == *b
+}
+
+// setResourceAdditionalFields will describe the fields that are not return by
+// DescribeTable calls
+func (rm *resourceManager) setResourceAdditionalFields(
+	ctx context.Context,
+	ko *v1alpha1.Table,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.setResourceAdditionalFields")
+	defer exit(err)
+
+	ko.Spec.Tags, err = rm.getResourceTagsPagesWithContext(ctx, string(*ko.Status.ACKResourceMetadata.ARN))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getResourceTagsPagesWithContext queries the list of tags of a given resource.
+func (rm *resourceManager) getResourceTagsPagesWithContext(ctx context.Context, resourceARN string) ([]*v1alpha1.Tag, error) {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.getResourceTagsPagesWithContext")
+	defer exit(err)
+
+	tags := []*v1alpha1.Tag{}
+
+	var token *string = nil
+	for {
+		var listTagsOfResourceOutput *svcsdk.ListTagsOfResourceOutput
+		listTagsOfResourceOutput, err = rm.sdkapi.ListTagsOfResourceWithContext(
+			ctx,
+			&svcsdk.ListTagsOfResourceInput{
+				NextToken:   token,
+				ResourceArn: &resourceARN,
+			},
+		)
+		rm.metrics.RecordAPICall("GET", "ListTagsOfResource", err)
+		if err != nil {
+			return nil, err
+		}
+		tags = append(tags, resourceTagsFromSDKTags(listTagsOfResourceOutput.Tags)...)
+		if listTagsOfResourceOutput.NextToken == nil {
+			break
+		}
+		token = listTagsOfResourceOutput.NextToken
+	}
+	return tags, nil
+}
```

Comment:
gotta love that `ListTagsForResource` has no `ListTagsForResourcePages` corresponding aws-sdk-go construct, like every other List operation does.... :/